code/src/routes/protected-route/ProtectedRoute.test.tsx

```diff
@@ -0,0 +1,91 @@
+import { screen } from "@testing-library/react";
+import { Navigate } from "react-router-dom";
+import {
+  useIsAuthLoadingSelector,
+  useIsAuthSelector,
+  useUserDetailsSelector
+} from "@/store/slices/userSlice";
+import ProtectedRoute from "@/routes/protected-route/ProtectedRoute";
+import { UserRole } from "@/types/user.types";
+import routePaths from "@/constants/routes";
+import { ROLES } from "@/constants/common";
+import renderWithProviders from "@/utils/render-with-providers/renderWithProviders";
+
+jest.mock("@/store/slices/userSlice", () => ({
+  useIsAuthLoadingSelector: jest.fn(),
+  useIsAuthSelector: jest.fn(),
+  useUserDetailsSelector: jest.fn()
+}));
+
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  Navigate: jest.fn(() => null)
+}));
+
+jest.mock("@/layouts/page-loading-fallback/PageLoadingFallback", () => ({
+  __esModule: true,
+  default: () => <div>Loading...</div>
+}));
+
+type RenderComponent = {
+  isAuthenticated: boolean;
+  isLoading: boolean;
+  role: UserRole;
+};
+
+const renderComponent = ({
+  isAuthenticated = false,
+  isLoading = false,
+  role
+}: Partial<RenderComponent> = {}) => {
+  (useIsAuthLoadingSelector as jest.Mock).mockReturnValue(isLoading);
+  (useIsAuthSelector as jest.Mock).mockReturnValue(isAuthenticated);
+  (useUserDetailsSelector as jest.Mock).mockReturnValue(role ? { role } : null);
+
+  return renderWithProviders(
+    <ProtectedRoute
+      element={<div>Protected Content</div>}
+      allowedRoles={[ROLES.ADMIN]}
+    />
+  );
+};
+
+describe("ProtectedRoute", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("shows loading fallback initially", () => {
+    renderComponent({ isLoading: true });
+
+    const loadingComponent = screen.getByText("Loading...");
+    expect(loadingComponent).toBeInTheDocument();
+  });
+
+  test("redirects to unauthorized path if user is not authenticated", () => {
+    renderComponent();
+
+    expect(Navigate).toHaveBeenCalledWith({ to: routePaths.home.path }, {});
```

Comment:
When the user is not authenticated in your title. Why it should be `home path` If we at the same page? 
And I suggest to use just `toHaveBeenCalled()` to avoid some questions about it

Reply:
I used `forbiddenRedirectPath` and `unauthorizedRedirectPath` instead and I hope it would be better to test if we actually redirect to passed path